src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
> Some clarification - this PR is not using the globally defined MAX value. It uses valueMetaData.dataLength, this is the data length derived from column header of the row.

Sorry I was not clear. I understand the code uses valueMetaData.dataLength for data length. What I'm referring is the globally used MAX, for example <code>if (metaData.dataLength === MAX) {</code> under <code>case 'Char'</code> in <code>valueParse</code>. Per spec it seems like for Variant case instead of MAX, the max length read from the stream should be used.

If SqlClient is ignoring that, I think it's fine if we're doing the same here.